src/Nest/CommonAbstractions/Infer/Id/IdExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Nest
+{
+    public static class IdExtensions
+    {
+		public static bool IsConditionless(this Id id)
```

Comment:
Should we make these extensions internal?  Not just this one in particular, but any `IsCondtionless()` extension (i.e. `TypeName.IsConditionless()`).